Script to increase c-clef octaves by 1;tested 94r

kyrieb-ekat opened the pull request.

kyrieb-ekat committed:
Script to increase c-clef octaves by 1;tested 94r

dchiller commented on update_octave.py:
```diff
@@ -0,0 +1,29 @@
+    changes_made = 0
+    
+    for elem in root_element.iter():
+        if elem.tag.endswith('}clef'):
```
I think this is the only line I'm confused about...why the close curly brace and the `endswith`? why not just check the tag itself?

kyrieb-ekat replied:
Ah! Because the tag itself includes the namespace```"{http://www.music-encoding.org/ns/mei}clef"```  (which you'd also get if you printed the ```elem.tag``` for a clef element. So checking for ```endswith('}clef')``` is a way to match the element regardless of its namespace. It's looking for any tag that ends with "}clef", basically. Probably everything in the file uses the one flavor of MEI, but this way it's a little more flexible and any element with ```}clef``` would be grabbed.

dchiller replied:
> a way to match the element regardless of its namespace

True, though I think in general that's probably a bug rather than a feature... we really do mean an MEI `clef` here and not some other clef. But since this is a one-and-done on a known set of files, sounds good to me.

dchiller commented on resulting_mei_files/Einsiedeln/Reviewed_once/CH-E_611_094r.mei.updated:
```diff
@@ -0,0 +1,2962 @@
+<?xml version='1.0' encoding='utf-8'?>
+<ns0:mei xmlns:ns0="http://www.music-encoding.org/ns/mei" meiversion="5.0.0-dev">
```
looks like the namespace has changed from the original? thoughts?

kyrieb-ekat replied:
I think this is to do with how ElementTree writes XML, wherein it needs to represent namespace and creates a prefix; in this case ```ns0``` (this is stemming from my very foundational understanding of how this works, so I may be mistaken). If you think its better to hold onto the original, adding something like ```ET.register_namespace('', 'http://www.music-encoding.org/ns/mei')``` shouldn't be too tricky? Also might be the wrong way forward, keen to hear your instinct/vibe. I don't believe the altered namespace should alter the validity of the MEI, however.

dchiller replied:
Oh, I see now. Yeah, it seems cleaner to just do what was in the original which is have the mei namespace be the default.

dchiller commented on resulting_mei_files/Einsiedeln/Reviewed_once/CH-E_611_094r.mei.updated:
```diff
@@ -0,0 +1,2962 @@
+                <ns0:score xml:id="m-7f90b96a-e0ba-4721-97e0-9153a664bab9">
+                    <ns0:scoreDef xml:id="m-e942e7c6-dd42-4e6d-a996-6f9d8e2ced2e">
+                        <ns0:staffGrp xml:id="m-a745669f-3ec6-4765-bf57-428ff6e6bb37">
+                            <ns0:staffDef xml:id="m-391f7576-9ce7-43e7-8fdb-222e259e7ff1" n="1" notationtype="neume" lines="4" clef.shape="C" clef.line="3" />
+                        </ns0:staffGrp>
+                    </ns0:scoreDef>
+                    <ns0:section xml:id="m-8f95eed5-16e8-4369-b06d-dc33e684b717">
+                        <ns0:staff n="1">
+                            <ns0:layer n="1">
+                                <ns0:sb n="1" facs="#m-59927400-6ec0-4ffe-a8c3-f4d6d8d6855c" xml:id="m-66e84da7-0b00-4624-9fbc-b7ce170e2755" />
+                                <ns0:clef xml:id="m-27d7665a-d8d8-4ad5-aeb0-8814f52a25c9" facs="#m-8d14e291-7247-4fd6-8ab7-c1dd66a12e62" shape="C" line="4" />
```
Do you know why the position of the clef on the `StaffDef` (line 891) is different than in the `clef` element (line 898)? 

It is like this in the original, so it's not a script issue, but it is weird.... The `clef` attribute is correct (the C clef is on line 4).

kyrieb-ekat replied:
I'm not sure- maybe it's a result of the original hard coding of the clef line (might be incorrect on where the original issue was set, in the ```clef``` element or ```staffDef```)? If the original location of the ```clef.line="3"``` was in ```staffDef``` maybe that's why?

yinanazhou replied:
`StaffDef` defines the default clef when there is no clef found on a `staff`. In the most recent files, `StaffDef@clef.line` should already be changed to 4 tho. Is it better to change it here or in another separate PR?

dchiller replied:
Ah ok, this makes sense, thanks. I don't know that we need to change this at all, since the issue we're dealing with here is the interpretation of clef octave, and not about what line it is on. If it is just a default, then this makes sense :)

dchiller commented on resulting_mei_files/Einsiedeln/Reviewed_once/CH-E_611_094r.mei.updated1:
```diff
@@ -0,0 +1,2962 @@
+<?xml version='1.0' encoding='utf-8'?>
+<ns0:mei xmlns:ns0="http://www.music-encoding.org/ns/mei" meiversion="5.0.0-dev">
```
It also looks like we lose the schema information that are in the original  xml files. See https://github.com/DDMAL/e2e-omr-resources/blob/5e9aaf692cb1614c5cded89ac2ea8c11d054015f/resulting_mei_files/Einsiedeln/Reviewed_once/CH-E_611_001r.mei#L2-L3

kyrieb-ekat replied:
Yikes! Will more explicitly snag the header!

kyrieb-ekat committed:
Fix: resolved header mismatch and namespace issues

kyrieb-ekat committed:
fix: MEI script now correctly updates octaves and
preserves file format
- Fixes file duplication issue in output
- Correctly increases octaves in C-clef sections
- Maintains proper MEI header formatting
- Preserves original XML structure

kyrieb-ekat committed:
Removed accidental duplicate file

kyrieb-ekat commented:
New test is done on 100v, removed other duplicate test files.

dchiller commented:
Ok, I think this PR is basically ready to merge. Just a few things:

1. I think we get rid of the test files. 
2. It looks like the test files have correct octaves on the custodes, but I think that's because you used files where that had previously been updated...so we need this script to also update custodes octaves.
3. For use on all of these files we should adjust input/output interfaces so we aren't typing each file name. 

Happy for you to take these on @kyrieb-ekat, or I can update this PR with them.

kyrieb-ekat commented:
>I think we get rid of the test files.
>It looks like the test files have correct octaves on the custodes, but I think that's because you used files where that had previously been updated...so we need this script to also update custodes octaves.
>For use on all of these files we should adjust input/output interfaces so we aren't typing each file name.
> 
> 
> Happy for you to take these on @kyrieb-ekat, or I can update this PR with them.

If it's okay for this to take a moment (~during/after reading week), I'm happy to do this, otherwise if you have the bandwidth and also there's a want to get this up quickly/over and done with, just let me know how I can assist! 

Re: custodes, I reckon that the manually changed files are also the files which had `dis= and dis.place` edits, which should help in at least finding all of them? An option which would need a little poking around, but would accelerate testing what we have or otherwise looping through those to knock everything out. Regardless, even without a specific search, shouldn't be a problem, I think.

dchiller commented:
Ok, I don't think it's a critical rush? So I will feel empowered but not obligated to take a crack at it if I find myself with downtime
